core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerService.js

```diff
@@ -199,14 +201,15 @@ oppia.factory('CollectionLinearizerService', [
        * collection.
        */
       removeCollectionNode: function(collection, explorationId) {
+        if (!collection.containsCollectionNode(explorationId)) {
+          return false;
+        }
+
         // Relinking is only needed if more than just the specified node is in
         // the collection.
         if (collection.getCollectionNodeCount() > 1) {
           var linearNodeList = _getCollectionNodesInPlayableOrder(collection);
           var nodeIndex = _findNodeIndex(linearNodeList, explorationId);
```

Comment:
Remove underscore and don't forget to modify `_findNodeIndex` function name in where it is defined.